src/jsx.d.ts

```diff
@@ -1320,9 +1326,315 @@ export namespace JSXInternal {
+		 * @see aria-rowindex.
+		 */
+		'aria-rowcount'?: Signalish<number | undefined>;
+		/**
+		 * Defines an element's row index or position with respect to the total number of rows within a table, grid, or treegrid.
+		 * @see aria-rowcount @see aria-rowspan.
+		 */
+		'aria-rowindex'?: Signalish<number | undefined>;
+		/**
+		 * Defines the number of rows spanned by a cell or gridcell within a table, grid, or treegrid.
+		 * @see aria-rowindex @see aria-colspan.
+		 */
+		'aria-rowspan'?: Signalish<number | undefined>;
+		/**
+		 * Indicates the current "selected" state of various widgets.
+		 * @see aria-checked @see aria-pressed.
+		 */
+		'aria-selected'?: Signalish<Booleanish | undefined>;
+		/**
+		 * Defines the number of items in the current set of listitems or treeitems. Not required if all elements in the set are present in the DOM.
+		 * @see aria-posinset.
+		 */
+		'aria-setsize'?: Signalish<number | undefined>;
+		/** Indicates if items in a table or grid are sorted in ascending or descending order. */
+		'aria-sort'?: Signalish<
+			'none' | 'ascending' | 'descending' | 'other' | undefined
+		>;
+		/** Defines the maximum allowed value for a range widget. */
+		'aria-valuemax'?: Signalish<number | undefined>;
+		/** Defines the minimum allowed value for a range widget. */
+		'aria-valuemin'?: Signalish<number | undefined>;
+		/**
+		 * Defines the current value for a range widget.
+		 * @see aria-valuetext.
+		 */
+		'aria-valuenow'?: Signalish<number | undefined>;
+		/** Defines the human readable text alternative of aria-valuenow for a range widget. */
+		'aria-valuetext'?: Signalish<string | undefined>;
+	}
+
+	// All the WAI-ARIA 1.1 role attribute values from https://www.w3.org/TR/wai-aria-1.1/#role_definitions
+	type AriaRole =
+		| 'alert'
+		| 'alertdialog'
+		| 'application'
+		| 'article'
+		| 'banner'
+		| 'button'
+		| 'cell'
+		| 'checkbox'
+		| 'columnheader'
+		| 'combobox'
+		| 'complementary'
+		| 'contentinfo'
+		| 'definition'
+		| 'dialog'
+		| 'directory'
+		| 'document'
+		| 'feed'
+		| 'figure'
+		| 'form'
+		| 'grid'
+		| 'gridcell'
+		| 'group'
+		| 'heading'
+		| 'img'
+		| 'link'
+		| 'list'
+		| 'listbox'
+		| 'listitem'
+		| 'log'
+		| 'main'
+		| 'marquee'
+		| 'math'
+		| 'menu'
+		| 'menubar'
+		| 'menuitem'
+		| 'menuitemcheckbox'
+		| 'menuitemradio'
+		| 'navigation'
+		| 'none'
+		| 'note'
+		| 'option'
+		| 'presentation'
+		| 'progressbar'
+		| 'radio'
+		| 'radiogroup'
+		| 'region'
+		| 'row'
+		| 'rowgroup'
+		| 'rowheader'
+		| 'scrollbar'
+		| 'search'
+		| 'searchbox'
+		| 'separator'
+		| 'slider'
+		| 'spinbutton'
+		| 'status'
+		| 'switch'
+		| 'tab'
+		| 'table'
+		| 'tablist'
+		| 'tabpanel'
+		| 'term'
+		| 'textbox'
+		| 'timer'
+		| 'toolbar'
+		| 'tooltip'
+		| 'tree'
+		| 'treegrid'
+		| 'treeitem'
+		| (string & {}); // This trick allows any string to be valid here but also provides TS auto-complete behavior for the explicitly listed values here
```

Comment:
What's the reason for allowing any string to be valid? Are there use cases for using roles outside of this list? Genuinely curious.

Reply:
Tbh, I'm not entirely sure. I've copied this from React. I assume it's to accommodate new roles as the spec changes? I need to check the git history of their types to see the reasoning.

Reply:
I couldn't think of a reason and went back through the aria spec and didn't find anything concrete there either. 

The spec doesn't change very quickly (last version was published December 2017) so having that carve out doesn't seem too pressing IMO. (That said, ARIA 1.2 should be published _soon_ but it's also been soon for like a year now... But that still a 6 year cadence if it gets published this year.) It's also kind of a slippery slope where you could argue any of the attributes should allow for anything because the spec could change.

Reply:
Ah, sorry, missed that. [Doesn't look like a reason was given](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/53361), unless I'm missing part of the conversation.

I'm admittedly not a TS maximalist, so I personally have no issue with a string fallback (working auto-complete is lovely enough), but I wonder if it's better to be strict and maybe catch some typos, even if that means we might need to make alterations in the future? I'm not super knowledgeable about aria stuff though, so maybe there is a use for other roles.

Reply:
@myasonik Yea, that makes sense. No point to reduce the usefulness of these typings just cuz anything could change lol. And like you said, the churn is super infrequent.

Thanks for digging that link up @rschristian! I dug a little further and it looks like this was the original thinking ([link](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/53344#pullrequestreview-666294951)):
> 1. It should still allow unknown roles e.g. when new roles are added but types are outdated
> 2. what about fallback roles e.g. role="none presentation"?

For 1, like @myasonik was saying, I'm not worried about it anymore. But the second reason is news to me. I've never heard of "fallback roles" before. Scanning the spec, I can only find [one reference to fallbacks roles in a note](https://www.w3.org/TR/wai-aria-1.1/#role-none-note-none).

Maybe to start off we can just manually add "none presentation" as a possible role value? And in the future deal with other fallback roles as changes to the spec land? At that time we can either manually add more fallback combinations or look at more advanced TS techniques if it becomes complicated.

Reply:
I went with adding `'none presentation'` for now as that's my vote, but would like to hear what others think as well before merging.

Reply:
Ah, I forgot about that... It's pretty uncommon but, yeah, `role` takes space separated list of roles that you can use as fallbacks (https://www.w3.org/TR/wai-aria-1.1/#host_general_role)

The `none presentation` example given in the spec was just given because `none` was new at the time so `presentation` was an appropriate fallback. But you can theoretically do it with any combination at any arbitrary length. 

Though I can't imagine this being used very often, I do think allowing everything the spec allows is probably important enough to allow `(string & {})`.